universal-application-tool-0.0.1/app/views/admin/programs/ProgramIndexView.java

```diff
@@ -39,14 +41,31 @@ public Content render(ImmutableList<ProgramDefinition> programs, Http.Request re
                 h1("All Programs").withClasses(Styles.MY_4),
                 each(
                     programs.stream()
-                        .sorted(Comparator.comparing(program -> program.lifecycleStage()))
-                        .collect(ImmutableList.toImmutableList()),
-                    e -> this.renderProgramListItem(e, request)),
-                renderNewProgramButton());
+                        .collect(
+                            new GroupByKeyCollector<>(
+                                programDefinition -> programDefinition.name())),
+                    e -> this.renderProgramListItem(e.getKey(), e.getValue(), request)),
+                renderNewProgramButton(),
+                maybeRenderPublishButton(programs, request));
 
     return layout.render(head(layout.tailwindStyles()), body(contentDiv));
   }
 
+  private Tag maybeRenderPublishButton(
```

Comment:
(blocking) it would be helpful to state this only renders if there are DRAFT programs.

Reply:
Added.